app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/NotificationFilterActivity.java

```diff
@@ -0,0 +1,214 @@
+    public static final int NOTIFICATION_FILTER_MODE_WHITELIST = 1;
+    public static final int NOTIFICATION_FILTER_MODE_BLACKLIST = 2;
+    public static final int NOTIFICATION_FILTER_SUBMODE_ANY = 0;
+    public static final int NOTIFICATION_FILTER_SUBMODE_ALL = 1;
+
+    private Button mButtonSave;
+    private Spinner mSpinnerFilterMode;
+    private Spinner mSpinnerFilterSubMode;
+    private NotificationFilter mNotificationFilter;
+    private EditText mEditTextWords;
+    private DBHandler db = null;
+    private List<String> mWordsList = new ArrayList<>();
+    private List<Long> mFilterEntryIds = new ArrayList<>();
+
+    private static final Logger LOG = LoggerFactory.getLogger(NotificationFilterActivity.class);
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_notification_filter);
+
+        String packageName = getIntent().getStringExtra(AppBlacklistAdapter.STRING_EXTRA_PACKAGE_NAME);
+
+        if (StringUtils.isBlank(packageName)) {
+            this.finish();
+        }
+
+        packageName = packageName.toLowerCase();
+
+        try {
+            db = GBApplication.acquireDB();
+        } catch (GBException e) {
+            LOG.error("Could not acquire DB.", e);
+            this.finish();
+        }
+
+        NotificationFilterDao notificationFilterDao = db.getDaoSession().getNotificationFilterDao();
+        NotificationFilterEntryDao notificationFilterEntryDao = db.getDaoSession().getNotificationFilterEntryDao();
+
+        Query<NotificationFilter> query = notificationFilterDao.queryBuilder().where(NotificationFilterDao.Properties.AppIdentifier.eq(packageName)).build();
+        mNotificationFilter = query.unique();
+
+        if (mNotificationFilter == null) {
+            mNotificationFilter = new NotificationFilter();
+            mNotificationFilter.setAppIdentifier(packageName);
+            LOG.debug("New Notification Filter");
+        } else {
+            LOG.debug("Loaded existing notification filter");
+            Query<NotificationFilterEntry> queryEntries = notificationFilterEntryDao.queryBuilder().where(NotificationFilterEntryDao.Properties.NotificationFilterId.eq(mNotificationFilter.getId())).build();
+            List<NotificationFilterEntry> filterEntries = queryEntries.list();
+            if (!filterEntries.isEmpty()) {
+                for (NotificationFilterEntry temp : filterEntries) {
+                    mWordsList.add(temp.getNotificationFilterContent());
+                    mFilterEntryIds.add(temp.getId());
+                    LOG.debug("Loaded filter word: " + temp.getNotificationFilterContent());
+                }
+            }
+        }
+
+        setupView();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+
+        if (db != null) {
+            GBApplication.releaseDB();
```

Comment:
If you can, please use the try-with-resources pattern to acquire and release the db. Is there a special reason why you keep the lock during the entire activity lifetime? I think you can release it (automatically with try-with-resources) in `onCreate()`, so it should even be local variable instead of a field.